src/main/java/com/echobox/api/linkedin/connection/OrganizationConnection.java

```diff
@@ -0,0 +1,407 @@
+   * community-management/organizations/share-statistics">
+   * Organization Share Statistics</a>
+   * @see <a href="https://docs.microsoft.com/en-us/linkedin/marketing/integrations/
+   * community-management/organizations/share-statistics#retrieve-lifetime-share-statistics">
+   * Organization Share Statistics - Lifetime</a>
+   * @see <a href="https://docs.microsoft.com/en-us/linkedin/marketing/integrations/
+   * community-management/organizations/share-statistics#retrieve-time-bound-share-statistics">
+   * Organization Share Statistics - time-bound</a>
+   * @see <a href="https://docs.microsoft.com/en-us/linkedin/marketing/integrations/
+   * community-management/organizations/share-statistics#retrieve-statistics-for-specific-shares">
+   * Organization Share Statistics - specific shares</a>
+   * @param organizationURN the ogranization URN to retrieve the share statistics
+   * @param timeInterval the time interval for time-bounded statistics
+   * @param shareURNs specific share URNs
+   * @return List of organization share statistics
+   */
+  public List<Statistics> retrieveOrganizationShareStatistics(URN organizationURN,
+      TimeInterval timeInterval, List<URN> shareURNs) {
+    validateOrganizationURN("organizationURN", organizationURN);
+
+    List<Parameter> parameters = new ArrayList<>();
+    parameters.add(Parameter.with(QUERY_KEY, ORGANIZATION_VALUE));
+    parameters.add(Parameter.with(ORGANIZATIONAL_ENTITY_KEY, organizationURN.toString()));
+
+    addTimeIntervalToQueryParameters(timeInterval, parameters);
+
+    if (shareURNs != null && !shareURNs.isEmpty()) {
+      shareURNs.stream().forEach(this::validateShareURN);
+      parameters.add(Parameter.with(SHARES_KEY, shareURNs));
+    }
+
+    return getListFromQuery(ORGANIZATION_ENTITY_SHARE_STATS, Statistics.class,
+        parameters.toArray(new Parameter[parameters.size()]));
+  }
+
+  private void validateOrganizationURN(String paramName, URN organizationURN) {
+    ValidationUtils.verifyParameterPresence(paramName, organizationURN);
+    if (!URNEntityType.ORGANIZATION.equals(organizationURN.getURNEntityType())) {
+      throw new IllegalArgumentException("The URN should be type ORGANIZATION");
+    }
+  }
+  
+  private void validateShareURN(URN shareURN) {
+    ValidationUtils.verifyParameterPresence("share", shareURN);
+    if (!URNEntityType.SHARE.equals(shareURN.getURNEntityType())) {
+      throw new IllegalArgumentException("The URN should be type SHARE");
+    }
+  }
+
+  private void addTimeIntervalToQueryParameters(TimeInterval timeInterval,
+      List<Parameter> parameters) {
+    if (timeInterval != null) {
+      if (!StringUtils.isBlank(timeInterval.getTimeGranularityType())) {
+        parameters.add(Parameter.with("timeIntervals.timeRange",
+            timeInterval.getTimeGranularityType()));
+      }
+      if (timeInterval.getTimeRange() != null && timeInterval.getTimeRange().getStart() != null
```

Comment:
Start or end can be null but not at the same time. Do we want to include this case?